src/core/thread/mle.hpp

```diff
@@ -602,6 +602,19 @@ class Mle : public InstanceLocator
      */
     bool IsAttached(void) const;
 
+    /**
+     * This method indicates whether device is currently attaching or not.
+     *
+     * Note that an already attached device may also be in attaching state. Examples of this include a leader/router
+     * trying to attach to a better partition, or a child trying to find a better parent (when feature
+     * `OPENTHREAD_CONFIG_PARENT_SEARCH_ENABLE` is enabled).
+     *
+     * @retval TRUE   Device is currently trying to attach.
+     * @retval FALSE  Device is not in middle of attach process.
+     *
+     */
+    bool IsAttaching(void) const { return (mAttachState != kAttachStateIdle); }
```

Comment:
How about using this method here:
https://github.com/openthread/openthread/blob/c740a053fd3181c085a7e13a2f4c6a66ed413b17/src/core/thread/mle.cpp#L627